python/cudf/cudf/core/dataframe.py

```diff
@@ -5040,82 +5041,83 @@ def isin(self, values):
         falcon      True       True
         dog        False      False
         """
-        if isinstance(values, dict):
-
-            result_df = DataFrame()
-
-            for col in self._data.names:
-                if col in values:
-                    val = values[col]
-                    result_df[col] = self._data[col].isin(val)
-                else:
-                    result_df[col] = column.full(
-                        size=len(self), fill_value=False, dtype="bool"
-                    )
-
-            result_df.index = self.index
-            return result_df
-        elif isinstance(values, Series):
+        # TODO: propagate nulls through isin
+        # https://github.com/rapidsai/cudf/issues/7556
+
+        def make_false_column_like_self():
+            return column.full(size=len(self), fill_value=False, dtype="bool")
+
+        # Preprocess different input types into a mapping from column names to
+        # a list of values to check.
+        result = {}
+        if isinstance(values, (Series, DataFrame)):
```

Comment:
aren't these the only two kinds of `IndexedFrame` ?
```suggestion
        if isinstance(values, IndexedFrame):
```

Reply:
Yep good call.